src/oc/web/actions/activity.cljs

```diff
@@ -239,13 +242,30 @@
       (dis/dispatch! [:modal-editing-activate]))
     (dis/dispatch! [:modal-editing-deactivate])))
 
+
+(declare entry-save)
+
 (defn entry-save-on-exit
-  [edit-key activity-data entry-body]
+  [edit-key activity-data entry-body section-editing]
   (let [entry-map (assoc activity-data :body entry-body)
         cache-key (get-entry-cache-key (:uuid activity-data))]
     (uc/set-item cache-key entry-map
      (fn [err]
        (when-not err
+         (when (and (not= "published" (:status activity-data))
+                    (:has-changes activity-data)
+                    (not (:auto-saving activity-data)))
+           ;; dispatch that you are auto saving
+           (dis/dispatch! [:update [edit-key]
+                           #(merge % activity-data {:auto-saving true})])
+           (entry-save edit-key activity-data section-editing
```

Comment:
Saving the entry with the server on each edit can be a problem because on the first edit we don't have a uuid and after the first save we do. This will compromise the local save to cache that is based on the entry uuid.
In the entry-save callback we need to check if the entry had a uuid, if not remove the cached item and save a new one with the proper uuid.

Reply:
Also check that the revision-id is properly saved after you start editing a post without a uuid and it's saved the first time, as i mentioned in Slack i started a new post, edited it 3 times, closed out of the edit and the revert was for version 1 instead of 0 (probably because after the first save it replaced the starting revision).